lib/tidewave/mcp/tools/eval.ex

```diff
@@ -120,25 +129,28 @@ defmodule Tidewave.MCP.Tools.Eval do
     end
   end
 
-  defp eval_with_captured_io(code, inspect_opts) do
-    result =
+  defp eval_with_captured_io(code, arguments, inspect_opts) do
+    {result, io} =
       capture_io(fn ->
         IOForwardGL.with_forwarded_io(:standard_error, fn ->
           try do
-            {result, _bindings} = Code.eval_string(code, [], env())
+            {result, _bindings} = Code.eval_string(code, [arguments: arguments], env())
             result
           catch
             kind, reason -> Exception.format(kind, reason, __STACKTRACE__)
           end
         end)
       end)
 
-    case result do
-      # this is returned by IEx helpers
-      {:"do not show this result in output", io} -> io
-      {result, ""} -> inspect(result, inspect_opts)
-      {result, io} -> "IO:\n\n#{io}\n\nResult:\n\n#{inspect(result, inspect_opts)}"
-    end
+    Jason.encode!(%{
```

Comment:
Should we have an argument whether the output is JSON or not? Then we return text by default, but we could pass true from TC if we need something more structured.

Reply:
Outside of TC it would make more sense to return text, and given how central `project_eval` is, it may be worth it. Your call :)